sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -85,6 +86,11 @@ case class CreateTableAsSelectExec(
     val table = catalog.createTable(
       ident, getV2Columns(query.schema, catalog.useNullableQuerySchema),
       partitioning.toArray, properties.asJava)
+    // Check if the table schema matches the schema of the query.
```

Comment:
shall we do this check in a lower level? where we instantiate the v2 Table.

Reply:
Make sense